boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -754,3 +804,159 @@ TEST_F(
+        App_CanMsgs_dcm_critical_errors_pack,
+        App_SharedErrorTable_GetBoardsWithCriticalErrors,
+        App_SharedErrorTable_GetAllCriticalErrors);
+}
+
+TEST_F(SharedErrorTableTest, process_dim_non_critical_errors)
+{
+    // A list of DIM non-critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> dim_non_critical_error_ids = {
+        DIM_NON_CRITICAL_ERRORS
+    };
+
+    TestRoutineForSetErrorsFromCanMsg(
+        DIM, dim_non_critical_error_ids,
+        CANMSGS_DIM_NON_CRITICAL_ERRORS_FRAME_ID,
+        CANMSGS_DIM_NON_CRITICAL_ERRORS_LENGTH,
+        App_CanMsgs_dim_non_critical_errors_pack,
+        App_SharedErrorTable_GetBoardsWithNonCriticalErrors,
+        App_SharedErrorTable_GetAllNonCriticalErrors);
+}
+
+TEST_F(SharedErrorTableTest, process_dim_critical_errors)
+{
+    // A list of DIM critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> dim_critical_error_ids = { DIM_CRITICAL_ERRORS };
+
+    TestRoutineForSetErrorsFromCanMsg(
+        DIM, dim_critical_error_ids, CANMSGS_DIM_CRITICAL_ERRORS_FRAME_ID,
+        CANMSGS_DIM_CRITICAL_ERRORS_LENGTH,
+        App_CanMsgs_dim_critical_errors_pack,
+        App_SharedErrorTable_GetBoardsWithCriticalErrors,
+        App_SharedErrorTable_GetAllCriticalErrors);
+}
+
+TEST_F(SharedErrorTableTest, process_fsm_non_critical_errors)
+{
+    // A list of FSM non-critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> fsm_non_critical_error_ids = {
+        FSM_NON_CRITICAL_ERRORS
+    };
+
+    TestRoutineForSetErrorsFromCanMsg(
+        FSM, fsm_non_critical_error_ids,
+        CANMSGS_FSM_NON_CRITICAL_ERRORS_FRAME_ID,
+        CANMSGS_FSM_NON_CRITICAL_ERRORS_LENGTH,
+        App_CanMsgs_fsm_non_critical_errors_pack,
+        App_SharedErrorTable_GetBoardsWithNonCriticalErrors,
+        App_SharedErrorTable_GetAllNonCriticalErrors);
+}
+
+TEST_F(SharedErrorTableTest, process_fsm_critical_errors)
+{
+    // A list of FSM critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> fsm_critical_error_ids = { FSM_CRITICAL_ERRORS };
+
+    TestRoutineForSetErrorsFromCanMsg(
+        FSM, fsm_critical_error_ids, CANMSGS_FSM_CRITICAL_ERRORS_FRAME_ID,
+        CANMSGS_FSM_CRITICAL_ERRORS_LENGTH,
+        App_CanMsgs_fsm_critical_errors_pack,
+        App_SharedErrorTable_GetBoardsWithCriticalErrors,
+
+        App_SharedErrorTable_GetAllCriticalErrors);
+}
+
+TEST_F(SharedErrorTableTest, process_pdm_non_critical_errors)
+{
+    // A list of PDM non-critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> pdm_non_critical_error_ids = {
+        PDM_NON_CRITICAL_ERRORS
```

Comment:
Instead of hard-coding the error enums by-hand, we now use the auto-generated macro (which expands to enums). This way, when we add/remove errors from the DBC we don't have to manually update this test case as well. The same applies to other boards.